fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/logical/LogicalSubQueryAlias.java

```
@@ -0,0 +1,113 @@
+        this(tableAlias, Optional.empty(), Optional.empty(), child);
+    }
+
+    public LogicalSubQueryAlias(String tableAlias, Optional<GroupExpression> groupExpression,
+            Optional<LogicalProperties> logicalProperties, CHILD_TYPE child) {
+        super(PlanType.LOGICAL_SUBQUERY_ALIAS, groupExpression, logicalProperties, child);
+        this.alias = tableAlias;
+
+    }
+
+    public List<Slot> computeOutput(Plan input) {
+        return input.getOutput().stream()
+                .map(slot -> slot.withQualifier(ImmutableList.of(alias)))
+                .collect(Collectors.toList());
+    }
+
+    public String getAlias() {
+        return alias;
+    }
+
+    @Override
+    public String toString() {
+        return "LogicalSubQueryAlias (" + alias.toString() + ")";
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        LogicalSubQueryAlias that = (LogicalSubQueryAlias) o;
+        return alias.equals(that.alias);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(alias);
+    }
+
+    @Override
+    public Plan withChildren(List<Plan> children) {
+        Preconditions.checkArgument(children.size() == 1);
+        return new LogicalSubQueryAlias<>(alias, children.get(0));
+    }
+
+    @Override
+    public <R, C> R accept(PlanVisitor<R, C> visitor, C context) {
+        return visitor.visitSubQueryAlias((LogicalSubQueryAlias<Plan>) this, context);
+    }
+
+    @Override
+    public List<Expression> getExpressions() {
+        return Collections.emptyList();
+    }
+
+    @Override
+    public Plan withGroupExpression(Optional<GroupExpression> groupExpression) {
+        return new LogicalSubQueryAlias<>(alias, groupExpression, Optional.of(logicalProperties), child());
+    }
```

Will `LogicalSubQueryAlias`  still exist in the Memo? If not, I think it'd be better to not overwrite this method.

---

Will not, but it's not need to throw a exception as their is always a node below the node so the child() will never be null

---

Ok, thx